kotlin/semlang-parser/src/main/kotlin/net/semlang/validator/validator.kt

```diff
@@ -159,9 +159,13 @@ private class Validator(
         val arguments = validateArguments(function.arguments, function.typeParameters.associateBy(TypeParameter::name)) ?: return null
         val returnType = validateType(function.returnType, function.typeParameters.associateBy(TypeParameter::name)) ?: return null
 
+        if (returnType.isReference()) {
+            errors.add(Issue("Reference types cannot be returned from functions", function.returnTypeLocation, IssueLevel.ERROR))
```

Comment:
We might want this to refer solely to aliased reference types instead... though in this case, the question would be whether it's aliased from outside the function (i.e., it's passed in through an argument, or is passed out through a Var-of-reference-type, which maybe shouldn't be a thing we allow...) rather than the type of checking we're doing here.